avogadro/rendering/shader.cpp

```diff
@@ -41,7 +41,17 @@ bool Shader::compile()
     m_handle = 0;
   }
 
-  GLenum type_ = m_type == Vertex ? GL_VERTEX_SHADER : GL_FRAGMENT_SHADER;
+GLenum type_;
```

Comment:
This is probably better handled through a switch statement in C++

switch (m_type) {
case Vertex:
type_ = GL_VERTEX_SHADER;
break;
case Fragment:
type_ = GL_FRAGMENT_SHADER;
// etc.
}